src/renderer/components/+namespaces/namespace.store.ts

```diff
@@ -22,12 +22,17 @@ export const namespaceUrlParam = createPageParam<string[]>({
 export class NamespaceStore extends KubeObjectStore<Namespace> {
   api = namespacesApi;
   contextNs = observable.array<string>();
+  @observable metrics: IPodMetrics = null;
 
   constructor() {
     super();
     this.init();
   }
 
+  async loadMetrics(namespace: Namespace) {
+    this.metrics = await namespacesApi.getMetrics(namespace.getName(), "");
+  }
+
```

Comment:
Main reason for such a store method is neccessity to keep resolved metrics in `@observable metrics` variable which caused by `<ResouceMetrics>` component. It receives metrics as prop. So, currently we need to keep metrics variable in store or in the `<NamespaceDetails/>` component itself. I guess store is a better place because (if I remember correctly) from mobx documentation: 
> Everything that can derived from store, should be derived from store

But I agree that we can refactor `<ResouceMetrics/>` for eliminating store `loadMetrics` calls.

Reply:
My concern is that we seem to be pilling a lot onto these stores (especially `NamespaceStore`) and given that it is a public field of a class that is exported we cannot just go and remove it later.

Finally, it isn't really being derived from application state since we are calling the `loadMetrics` function on every single render.

I really think a better solution would be to have a metrics store that is query-able by resource type

Reply:
Originally I did this because it was how other components already worked. That said I completely agree that we shouldn't pollute all of these store classes -> I'll refactor this PR.

Reply:
Refactored metrics stuff to standalone functions. This way we don't pollute extension api.